connector.go

```diff
@@ -283,12 +283,14 @@ func (c *Connector) GetObject(obj IBObject, ref string, res interface{}) (err er
 	queryParams := QueryParams{forceProxy: false}
 	resp, err := c.makeRequest(GET, obj, ref, queryParams)
 	//to check empty underlying value of interface
-	var result []map[string]interface{}
+	var result interface{}
 	err = json.Unmarshal(resp, &result)
 	if err != nil {
 		log.Printf("Cannot unmarshall to check empty value '%s', err: '%s'\n", string(resp), err)
 	}
-	if resp == nil || len(result) == 0 {
+
+        var data []interface{}
+	if resp == nil || (reflect.TypeOf(result) == reflect.TypeOf(data) && len(result.([]interface{})) == 0) {
```

Comment:
It's not clear what the reason is for the changes in GetObject(). Could you please elaborate?

Reply:
Because of forced proxy implementation , it was assumed that the result/response we get was either nil or list. but when a response which is of type dictionary was coming it was failing. this was done to  handle the case when a dictionary was coming wasn't implemented.

Reply:
```
it was assumed that the result/response we get was either nil or list
```
I'm afraid I don't understand. In the original code, result IS a map: `var result []map[string]interface{}`

BTW, can you also include example responses that failed in the original code please?

Reply:
I'm afraid it is still not clear.

IIRC, the format of the response is no different whether the request is proxied or not. I.e. the response is always something like:

```
   [
        {
            attr1: value,
            attr2: value,
               ...
          },
        {
            attr1: value,
            attr2: value,
               ...
          }
    ]
```
Or, if there's no result, then `[]`

So, in both cases, the `result` type `[]map[string]interface{}` should work.

Am I missing something? If you have encountered a case where it doesn't work, could you please show the response data - so we can discuss how best to address it?

Reply:
@yuewko  shall i go ahead and resolve this comment atleast ? based on the discussion

Reply:
@saiprasannasastry , from our offline conversation, it sounds like that the reason the original code doesn't could be because: when we're doing a GET by reference, it returns a single map - and therefore []map[string]{} wouldn't work.

If that is the case, you might want to check for whether ref is used for query, if so, pass interface{}(or map[string]interface) to unmarshall, otherwise pass []map[string]interface{} like the original code.

Otherwise, I'm OK as implemented.

Thanks!